crates/bevy_render/src/render_resource/resource_macros.rs

```diff
@@ -120,4 +120,29 @@ macro_rules! render_resource_wrapper {
     };
 }
 
+#[macro_export]
+macro_rules! define_atomic_id {
+    ($atomic_id_type:ident) => {
+        #[derive(Copy, Clone, Hash, Eq, PartialEq, Debug)]
+        pub struct $atomic_id_type(u32);
+
+        impl Default for $atomic_id_type {
+            fn default() -> Self {
+                use std::sync::atomic::{AtomicU32, Ordering};
+
+                static COUNTER: AtomicU32 = AtomicU32::new(0);
+                COUNTER
+                    .fetch_update(Ordering::Relaxed, Ordering::Relaxed, |val| {
+                        val.checked_add(1)
+                    })
+                    .map(Self)
+                    .expect(&format!(
```

Comment:
This always allocates, even when it's not panicking. `unwrap_or_else` with a panic inside may be what we actually want here.

Reply:
Good catch! I will change this.

Reply:
Done.